packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -322,8 +362,10 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
   ///
   /// The [maxLength] must be null, -1 or greater than zero. If it is null or -1
   /// then no limit is enforced.
-  LengthLimitingTextInputFormatter(this.maxLength)
-    : assert(maxLength == null || maxLength == -1 || maxLength > 0);
+  LengthLimitingTextInputFormatter(
+    this.maxLength, {
+    this.maxLengthEnforcement = MaxLengthEnforcement.enforced,
+  }) : assert(maxLength == null || maxLength == -1 || maxLength > 0);
```

Comment:
What if maxLength is null or -1, but maxLengthEnforcement is enforced?  I think that is ok, just checking...

Reply:
Then it should be the same as `maxLength: null` since the user use it in an incorrect way.
https://github.com/flutter/flutter/blob/95d276f943bdc4b9e007cefc6443e82e01f039ec/packages/flutter/lib/src/services/text_formatter.dart#L451-L455

Reply:
I think if (maxLength == null || maxLength < 0 || newValue.text.characters.length <= maxLength)

Reply:
why maxLength == -1?

Reply:
IMO -1 is same as null and opposite to n(n >= 0). The constructor already has an assertion that make sure maxLength is valid.

Reply:
It looks like adding that assert will break the current default (maxLength == null, maxLengthEnforced == true/maxLengthEnforced = .enforced)?

Reply:
@LongCatIsLooong That assert means?